### ISSUE TITLE
react/jsx-sort-props breaks the code

### ISSUE THREAD

**Issue by davesnx** [eslint/eslint#10881]
**Tell us about your environment**

    "babel-eslint": "^9.0.0",
    "eslint": "5.6.0",
    "eslint-plugin-react": "^7.11.1"

* **ESLint Version:** 9.0.0
* **Node Version:** 10.6.0
* **npm Version:** -
* **yarn Version:** 1.9.4

**What parser (default, Babel-ESLint, etc.) are you using?**
Babel-ESLint

**Please show your full configuration:**
I created a repo to reproduce the bug: https://github.com/davesnx/eslint-bug-react-sort-props

<details>
<summary>Configuration</summary>

```js
module.exports = {
  parser: 'babel-eslint',
  plugins: ['react'],
  rules: {
    'react/jsx-sort-props': 1
  }
}

```

</details>

**What did you do? Please include the actual source code causing the issue, as well as the command that you used to run ESLint.**

```js
import React, { Component } from 'react'
import PropTypes from 'prop-types'

const PictureChoice = () => {
  return (
    <List
      className={className}
      onStageAnswer={onStageAnswer}
      onCommitAnswer={onCommitAnswer}
      isFocused={isFocused}
      direction={direction}
      allowMultipleSelection={allowMultipleSelection}
      measureLongestChildNode={measureLongestChildNode}
      layoutItemsSize={layoutItemsSize}
      handleAppScroll={handleAppScroll}
      isActive={isActive}
      resetSelection={resetSelection}
      onKeyboardChoiceHovered={onKeyboardChoiceHovered}
      keyboardShortcutType
    />
  )
}

export default PictureChoice

```

```bash
npx eslint . --fix
```

**What did you expect to happen?**
So theorically the rule should re-order my props, in the worst case scenario, not sort them. But actually is breaking them. It's a weird output:

```js
import React, { Component } from 'react'
import PropTypes from 'prop-types'

const PictureChoice = () => {
  return (
    <List
      allowMultipleSelection={allowMultipleSelection}
      className={className}
      direction={direction}
      isFocused=handleAppScroll={handleAppScroll}ection=isActive={isActive}owMultiplisFocused={isFocused}ardShortcutType
      layoutItemsSize={layoutItemsSl
      layoutItemsSize={layoutItemsSize}
      handleAppScroll={handleAppScroll}
      isActive={isActive}
      resetSelection={resetSelection}
      onKeyboardChoiceHovered={onKeyboardChoiceHovered}
      keyboardShortcutType
    />
  )
}

export default PictureChoice
```

**Comment by davesnx** [comment]
It's actually trying to sort in a weird way, and it's not consistent.

```js

      isFocused=
      
      handleAppScroll={handleAppScroll}
      
      ection=
      
      isActive={isActive}
      
      owMultiplisFocused={isFocused}
      
      ardShortcutType
      layoutItemsSize={layoutItemsSl
      
```

**Comment by platinumazure** [comment]
Hi @davesnx, thanks for the issue.

This isn't a core ESLint rule. Do you have any evidence that core ESLint is causing this issue?

If not, could you please open this issue in the [eslint-plugin-react repository](https://github.com/yannickcr/eslint-plugin-react)? Thanks!

**Comment by davesnx** [comment]
I just opened on [eslint-plugin-react](https://github.com/yannickcr/eslint-plugin-react/issues/1993). But I can't use that rule without the plugin, but I got the feeling that had something to do with the parser or the AST being broken, this is why I opened here in the first place.

I'm happy to close it, tho.
Thanks @platinumazure

**Comment by platinumazure** [comment]
@davesnx It's highly unlikely this is a core ESLint issue.

Basically one of two things is happening:

1. babel-eslint might be producing an incorrect AST (with incorrect node range information) which could throw off the autofixer
1. The rule itself is producing bad autofix range information

I think it makes the most sense to start with eslint-plugin-react so they can see if the rule is incorrect, and then they might redirect you to babel-eslint if the parser ends up being the issue. Or, who knows, maybe there is a core issue after all :smile: But I think eslint-plugin-react should look at the rule first.

**Comment by ilyavolodin** [comment]
This looks like a bug in eslint plugin, not a bug in the eslint itself. Closing.
